lib/MusicBrainz/Server/Entity/Relationship.pm

```diff
@@ -339,7 +339,7 @@ around TO_JSON => sub {
     my $target_type = $self->target_type;
 
     my $json = {
-        attributes      => [map +{ (%{ $_->TO_JSON }, type => { gid => $_->type->gid }) }, $link->all_attributes],
+        attributes      => [map +{ (%{ $_->TO_JSON }, type => { id => $_->type->id, gid => $_->type->gid }) }, $link->all_attributes],
```

Comment:
No benefit, it's just that they're keyed by `id` in the `linked_entities` JSON that gets passed to Node, so it needs the `id` on the server. I agree that the `id` shouldn't really be public, but this is all internal.

It shouldn't affect seeding, because the code for that runs on the client where the linked entities are mapped by [both id and gid](https://github.com/metabrainz/musicbrainz-server/blob/3ba1376888d91cbca944314df0d0ea721e457e7a/root/static/scripts/relationship-editor/common/viewModel.js#L20-L28).

We might want to change `linked_entities` to key by `gid` when we convert the relationship editor to React, though, since at that point the seeding code may run on the server.